### PR TITLE
feat(skymp-client): use `startDeferredKill` to prevent unexpected death of any npc

### LINKED ISSUE #191: Use `startDeferredKill` to prevent unexpected death of any npc

### PR DESCRIPTION
**Pull request by ZikkeyLS**
Closes #191

### CODE REVIEW

**Review by alekcey0211 — CHANGES_REQUESTED** [review]
1. Check msg.props twice
2. i think curly brackets must be on one line with if statement

**Comment by ZikkeyLS** [comment]
That's basically test code, but I will fix and prettify it in next commits. Thanks for suggestions!

File: skymp5-client/src/front/model.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export interface FormModel {
   isOpen?: boolean;
   inventory?: Inventory;
   isHostedByOther?: boolean;
+  isMe?: boolean;
```
Comment (Pospelove) [inline skymp5-client/src/front/model.ts:21]: What is the purpose of this change?

File: skymp5-client/src/front/remoteServer.ts [hunk 2]
```diff
@@ -275,11 +276,14 @@ export class RemoteServer implements MsgHandler, ModelSource, SendTarget {
 
     // TODO: move to a separate module
 
-    if (msg.props && !msg.props.isHostedByOther) {
+    if(msg.props)
+    {
+      this.worldModel.forms[i].isHostedByOther = msg.props.isHostedByOther as boolean;
```
Comment (Pospelove) [inline skymp5-client/src/front/remoteServer.ts:281]: What is the purpose of this change? Consider not making unrelated changes.

File: skymp5-client/src/front/skympClient.ts [hunk 3]
```diff
@@ -133,7 +133,7 @@ export class SkympClient {
 
     on("update", () => {
       if (!this.singlePlayer) {
-        this.sendInputs();
+        this.updateData();
```
Comment (Pospelove) [inline skymp5-client/src/front/skympClient.ts:136]: Please keep the original name. `updateData` is too common and doesn't say anything.

File: skymp5-client/src/front/skympClient.ts [hunk 4]
```diff
@@ -447,21 +446,35 @@ export class SkympClient {
...
 
-  private sendInputs() {
+  private updateData()
+  {
     const hosted =
       typeof storage["hosted"] === typeof [] ? storage["hosted"] : [];
     const targets = [undefined].concat(hosted as any);
-    //printConsole({ targets });
+    // removing all stop hosted objects from usedRefID array
+    deathSystem.usedRefID.filter((value) => targets.includes(value))
```
Comment (Pospelove) [inline skymp5-client/src/front/skympClient.ts:455]: This line does not work as you expect. Filter doesn't modify the original array.  https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

File: skymp5-client/src/front/skympClient.ts [hunk 5]
```diff
@@ -447,21 +446,35 @@ export class SkympClient {
...
 
+  private prepareSendInputs(target : undefined) {
+    this.sendMovement(target);
+    this.sendAnimation(target);
+    this.sendLook(target);
+    this.sendEquipment(target);
+    this.sendActorValuePercentage(target);
+  }
+
+  private tryMakeActorImmortal(target : undefined)
```
Comment (Pospelove) [inline skymp5-client/src/front/skympClient.ts:473]: Making actors immortal is not related to the inputs we send. Please write the code that is responsible for **correctly show other characters** in `view.ts`

**Comment by Pospelove** [comment]
Two reviews are my mistake. The first set of comments is correct.

File: skymp5-client/src/front/deathSystem.ts [hunk 6]
```diff
@@ -24,6 +24,6 @@ hooks.sendAnimationEvent.add({
   },
 });
 
-export const MakePlayerImmortal = (): void => {
-  Game.getPlayer()?.startDeferredKill();
+export const MakeActorImmortal = (act: Actor): void => {
```
Comment (Pospelove) [inline skymp5-client/src/front/deathSystem.ts:27]: MakeActorImmortal  => makeActorImmortal

**Comment by alekcey0211** [comment]
I think you need to use `as` syntax in extreme cases. If TS show type error, you need to process it.

@Pospelove what do you think about this?

**Review by alekcey0211 — CHANGES_REQUESTED** [review]
1. MakeActorImmortal take Actor argument but why do you do the check ’act?’ Or you have the wrong type in the argument `act: Actor`
2. I prefer check like this 
```js
if (!player) {
  // some code
}
```
The code becomes more readable and in this case you add a check to ‘undefined’

Fix this and i approve it.

**Comment by ZikkeyLS** [comment]
Hopefully all stuff is correct now

**Review by alekcey0211 — CHANGES_REQUESTED** [review]
Form the `if` statement correctly. Put a space after `if` and curly brackets
```js
if (condition) {
 // ...
}
```
and formatting arrow functions like this. 
```js
() => {
  // ...
}
```

**Review by Pospelove — APPROVED** [review]

**Review by Pospelove — CHANGES_REQUESTED** [review]
Oops, you better fix formatting as alekcey0211 says

**Comment by ZikkeyLS** [comment]
Ready

**Comment by alekcey0211** [comment]
After `if` must be whitespace

**Review by Pospelove — CHANGES_REQUESTED** [review]
please `Right mouse button -> Format` in vs code

**Comment by ZikkeyLS** [comment]
OK, I will fix all in home

**Comment by alekcey0211** [comment]
Or use combination `shift + alt + f` to format code

**Comment by ZikkeyLS** [comment]
Format is ready!

**Review by Pospelove — APPROVED** [review]

**Pull request merged**